src/Http/ServerRequest.php

```diff
@@ -127,7 +127,7 @@ class ServerRequest implements ServerRequestInterface
         'delete' => ['env' => 'REQUEST_METHOD', 'value' => 'DELETE'],
         'head' => ['env' => 'REQUEST_METHOD', 'value' => 'HEAD'],
         'options' => ['env' => 'REQUEST_METHOD', 'value' => 'OPTIONS'],
-        'ssl' => ['env' => 'HTTPS', 'options' => [1, 'on']],
+        'https' => ['env' => 'HTTPS', 'options' => [1, 'on']],
```

Comment:
I wonder if instead of magic strings those should be class constants?
This way any rename would be visible through tools like PHPStan if in use etc.

Reply:
I also just thought of refactoring the whole detector related code into a `DetectorTrait` because ServerRequest is already quite big...

Reply:
I'm all for renaming ssl -> https. Could we also backport this change to 4.x? With a breaking change like this it costs us almost nothing to backport. This gives userland code a chance to upgrade before the change becomes breaking.

Let me know if you'd like me to do the backport. I'm more than happy to help with these kinds of requests.

Reply:
I think we'd need to support both in 4.x otherwise this is a very hidden change.

Reply:
Yes 4.x would have both.